HWM 82 Yes and No button disappear

TemitopeAyokuHO opened the pull request:
## What? 
[HWM-82](https://collaboration.homeoffice.gov.uk/jira/browse/HWM-82)

## Why? 
once user click on yes or no button it should disappear from screen 

## How? 
- added function to check if new message is a structured message and set visibity property to false
- added function to get the index of current structured message to allow send button utiilze the index once clicked.
- extended each functionality to each service
- wrote Unit test for each unctions

## Testing?
## Screenshots (optional)
## Anything Else? (optional)
## Check list

- [ ] I have reviewed my own pull request for linting issues (e.g. adding new lines)
- [ ] I have written tests (if relevant)
- [ ] I have created a JIRA number for my branch
- [ ] I have created a JIRA number for my commit
- [ ] I have followed the chris beams method for my commit https://cbea.ms/git-commit/
here is an [example commit](https://github.com/UKHomeOfficeForms/hof/commit/810959f391187c7c4af6db262bcd143b50093a6e)
- [ ] Ensure drone builds are green especially tests
- [ ] I will squash the commits before merging

TemitopeAyokuHO committed:
HWM 82 Yes and No button disappear

- added function to check if new message is a structured message and set visibity property to false

- added function to get the index of current structured message to allow send button utiilze the index once clicked.

- extended each functionality to each service

- wrote Unit test for each functions

TemitopeAyokuHO committed:
HWM 82 Yes and No button disappear

- added function to check if new message is a structured message and set visibity property to false

- added function to get the index of current structured message to allow send button utiilze the index once clicked.

- extended each functionality to each service

- wrote Unit test for each unctions

TemitopeAyokuHO committed:
HWM 82 Yes and No button disappear

- added function to check if new message is a structured message and set visibity property to false

- added function to get the index of current structured message to allow send button utiilze the index once clicked.

- extended each functionality to each service

- wrote Unit test for each unctions

gregwolversonHO commented on src/components/message/types/structured-message.js:
```diff
@@ -1,19 +1,37 @@
+
+  const handleKeyDown = (event, content) => {
+    if (event.key === "Enter") {
+      console.log("EVENT Key:", event.key)
```
Can remove console.log

gregwolversonHO commented on src/routes/eta.js:
```diff
@@ -75,6 +98,12 @@ export default function Eta() {
 
   }, []);
 
+  // useEffect(() => {
```
remove commented out code

gregwolversonHO reviewed (changes requested):
Some minor changes needed to existing code. However, you need to add tests to each service route (eta/euss/evisa) to cover the changes being made to the useEffects.

Ensure we have tests to cover the scenario where the user hasn't responded to a quick reply question and the bot sends another, only 1 set of buttons should be displayed.

gregwolversonHO committed:
HWM-69: implemented view cookies page (#33)

Implemented view cookies page as per JIRA: https://collaboration.homeoffice.gov.uk/jira/browse/HWM-69

gregwolversonHO committed:
HWM-66: implemented govuk rebrand for header and banner (#36)

Implemented the updates needed to bring the header and banner inline with the GDS rebrand. 

JIRA: https://collaboration.homeoffice.gov.uk/jira/browse/HWM-66

gregwolversonHO committed:
HWM-69: added missing cookies link in footer (#37)

Added missing cookies link to the Footer as per scenario 3 in JIRA: https://collaboration.homeoffice.gov.uk/jira/browse/HWM-69

gregwolversonHO committed:
HWM-69: added cookiePolicy to link state in footer (#38)

Implemented correct cookie policy in footer link to cookie pages

gregwolversonHO committed:
Feature/hwm 58 chat history (#34)

Implemented chat history as per requirements
* Refactored approach to subscribing to Genesys functions to ensure consistency
* Implemented restored subscription
* Implemented fetchHistory and historyComplete functions
* Added tests to cater for historical messages

JIRA: https://collaboration.homeoffice.gov.uk/jira/browse/HWM-58

TemitopeAyokuHO committed:
Merge branch 'main' into feature/HWM-82-Yes-and-No-button-disappear

gregwolversonHO reviewed (approved):
lgtm
